halibot/halobject.py

```diff
@@ -42,6 +42,21 @@ def init(self):
 	def shutdown(self):
 		pass
 
+	def apply_filter(self, dest):
+		fl = self._hal.config.get("filters")
+		if not fl:
+			return dest # Filters not enabled/configured
+
+		name = dest.split("/")[0]
+
+		ib = fl.get("inbound", {})
+		ret = "/".join(ib[name] + [dest]) if name in ib else dest
+
+		ob = fl.get("outbound", {})
+		ret = "/".join(ob[self.name] + [ret]) if self.name in ob else ret
```

Comment:
I'm growing increasingly displeased with verbatim string matching for RIs. Starting to think it might be worth adding an abstraction to handle RI comparison. Something where a check for `irc` will pass for both `irc` and `irc/##halibot`, with the former referring only to the agent (general case), the latter referring to a specific resource to/from `irc`.

As it stands, it occurred to me that this doesn't support per-channel filters, which might be useful for limiting output to a specific channel/user, rather than for all resources behind a particular agent. I can hack in support, but there's a deeper issue here I feel.

Reply:
I agree, we really shouldn't be spliting/joining on "/" everywhere. It definitely should be abstracted somewhere.

Reply:
Should we open a new issue to track this? I think this is fine for now, but we probably want some way to allow per-channel filtering as well.

Reply:
#135